dask/dataframe/io/sql.py

```diff
@@ -62,33 +72,37 @@ def read_sql_table(table, uri, npartitions, index_col, limits=None, columns=None
             minmax = pd.read_sql(q, engine)
             maxi, mini = minmax.iloc[0]
             if minmax.dtypes['max_1'].kind == "M":
-                npartitions = pd.date_range(
+                divisions = pd.date_range(
                     start=mini, end=maxi, freq='%iS' % (
                         (maxi - mini) / npartitions).total_seconds()).tolist()
+                divisions[-1] += datetime.timedelta(seconds=1)
             else:
-                npartitions = np.linspace(mini, maxi, npartitions + 1).tolist()
-                npartitions[-1] += 1
+                divisions = np.linspace(mini, maxi, npartitions + 1).tolist()
+                divisions[-1] += 1
         else:
             mini, maxi = limits
-            npartitions = np.linspace(mini, maxi, npartitions + 1).tolist()
-    columns = ([(table.columns[c] if isinstance(c, str) else c)
+            divisions = np.linspace(mini, maxi, npartitions + 1).tolist()
+    else:
+        divisions = npartitions
```

Comment:
I don't understand.  from_delayed (where this variable is ultimately headed) expects a tuple or list of values, not an integer.  I suspect that this line is not covered by tests.  I think that it would cause a failure.  

Also generally when I look at dask.dataframe code I always assume that `type(npartitions) is int` and `isinstance(divisions, (list, tuple, np.ndarray))`, much in the way that I assume `dsk` is a dict or `df` is a dataframe of some sort.  Breaking these conventions makes maintaining this code more expensive.